src/env0-environments-provider.ts

```diff
@@ -127,8 +96,75 @@ export class Env0EnvironmentsProvider
   > = this._onDidChangeTreeData.event;
 
   async refresh(): Promise<void> {
-    this.environments = await getEnvironmentsForBranch();
-    this._onDidChangeTreeData.fire();
+    if (this.isRefreshing) {
+      return;
+    }
+    this.isRefreshing = true;
+    try {
+      const newEnvironments = await getEnvironmentsForBranch();
+      if (this.shouldUpdate(newEnvironments)) {
+        showEnvironmentStatusChangedNotification(
+          this.environments,
+          newEnvironments
+        );
+        this.environments = newEnvironments;
+        this._onDidChangeTreeData.fire();
+      }
+    } finally {
+      this.isRefreshing = false;
+    }
+  }
+}
+
+function showEnvironmentStatusChangedNotification(
+  oldEnvironments: EnvironmentModel[],
+  newEnvironments: EnvironmentModel[]
+) {
+  for (const newEnvironment of newEnvironments) {
+    const envIndex: number = oldEnvironments.findIndex(
+      (env) => env.id === newEnvironment.id
+    );
+
+    if (envIndex === -1) {
+      continue;
+    }
+
+    if (
+      oldEnvironments[envIndex].updatedAt !== newEnvironment.updatedAt &&
+      oldEnvironments[envIndex].status !== newEnvironment.status
+    ) {
+      if (newEnvironment.status === "DEPLOY_IN_PROGRESS") {
```

Comment:
nit: consider using a `switch` instead the multiple `if`s

Reply:
17698a86ac60257a143b010a1983e3066ab1f45b